specs/Workers.md

```diff
@@ -0,0 +1,1228 @@
+        }
+    }
+```
+## Win32 C++
+```cpp
+    void ScenarioServiceWorkerManager::GetAllServiceWorkerRegistrations()
+    {
+        if (!m_serviceWorkerManager)
+        {
+            CreateServiceWorkerManager();
+        }
+        CHECK_FAILURE(m_serviceWorkerManager->GetServiceWorkerRegistrations(
+            Callback<ICoreWebView2GetServiceWorkerRegistrationsCompletedHandler>(
+                [this](
+                    HRESULT error, ICoreWebView2ServiceWorkerRegistrationCollectionView*
+                                      workerRegistrationCollection) -> HRESULT
+                {
+                  if(workerRegistrationCollection)
+                  {
+                      UINT32 workersCount = 0;
+                      CHECK_FAILURE(workerRegistrationCollection->get_Count(&workersCount));
+
+                      std::wstringstream message{};
+                      message << L"No of service workers registered: " << workersCount << std::endl;
+
+                      for (UINT32 i = 0; i < workersCount; i++)
+                      {
+                          Microsoft::WRL::ComPtr<ICoreWebView2ServiceWorkerRegistration>
+                              serviceWorkerRegistration;
+                          CHECK_FAILURE(workerRegistrationCollection->GetValueAtIndex(
+                              i, &serviceWorkerRegistration));
+
+                          wil::unique_cotaskmem_string scope;
+                          CHECK_FAILURE(serviceWorkerRegistration->get_Scope(&scope));
+
+                          message << L"Scope: " << scope.get();
+                      }
+
+                      m_appWindow->AsyncMessageBox(
+                          std::move(message.str()), L"Registered service workers");
+                  }
+
+                    return S_OK;
+                })
+                .Get()));
+    }
+```
+
+## Retrieving Service Worker Registration for a Specific Scope.
+
+The following example illustrates how to retrieve the `CoreWebView2ServiceWorkerRegistration` associated with a specific scope. This enables the host application to establish a communication channel with the service worker, facilitating the exchange of messages.
+
+## .NET/WinRT
+```c#
+    async void GetServiceWorkerRegisteredForScopeExecuted(object target, ExecutedRoutedEventArgs e)
+    {
+        var dialog = new TextInputDialog(
+            title: "Scope of the Service Worker Registration",
+            description: "Specify a scope to get the service worker",
+            defaultInput: "");
+        if (dialog.ShowDialog() == true)
+        {
+            try
+            {
+                if (ServiceWorkerManager_ != null)
+                {
+                    ServiceWorkerManager_ = WebViewProfile.ServiceWorkerManager;
+                }
+                CoreWebView2ServiceWorkerRegistration registration = await ServiceWorkerManager_.GetServiceWorkerRegistrationAsync(dialog.Input.Text);
```

Comment:
You should say somewhere in here what the format is of this string including an example.